### PR TITLE
refactor: add liveness probe to the operator

### PR DESCRIPTION
**Pull request by boranx**
Fix: https://issues.redhat.com/browse/OSD-7407

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -182,3 +190,7 @@ func addMetrics(ctx context.Context) {
 		log.Error(err, "Failed to configure OSD metrics")
 	}
 }
+
+func healthCheck(req *http.Request) error {
+	return nil
```
Comment (fahlmant) [inline cmd/manager/main.go:195]: Is there a case where we want to check the response here? Does this check basically just enable ping?

Comment (boranx) [reply cmd/manager/main.go:195]: for now, it's just enabling ping check which is similar to [healthz.Ping](https://github.com/kubernetes-sigs/controller-runtime/blob/v0.9.6/pkg/healthz/healthz.go#L186-L187) However, we can fill this with validations as a checker function

Comment (fahlmant) [reply cmd/manager/main.go:195]: Cool, makes sense

**Review by fahlmant — APPROVED** [review]
/lgtm
/approve
/hold

**Comment by boranx** [comment]
@fahlmant currently it's bound to metric servers port(8181) would we consider having it on a specific port different than that?

**Comment by fahlmant** [comment]
@boranx does the health check pass when testing? As long as the metrics server comes up fast enough then I think it's ok.

**Comment by boranx** [comment]
yup, that's what I see on my local
```shell
curl 127.0.0.1:8181/healthz
ok
```

**Comment by fahlmant** [comment]
I think it's a good starting place then. Feel free to unhold when ready

**Pull request merged**